### PR TITLE
feat(savers): refetch queries in <TransactionsProvider />

### PR DESCRIPTION
**Pull request by gomesalexandre**
## Description

This PR:

- refetches THORCHain savers queries at `<TransactionsProvider />` level vs. on status modal step previously
- adds an additional safety in not displaying DeFi actoin buttons when there are any queries matching the current DefiProvider (Savers) / DefiType (Staking) and query status we're targeting (pending)

## Notice

- [x] Have you followed the guidelines in our [Contributing]('https://github.com/shapeshift/web/blob/develop/CONTRIBUTING.md) guide?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/shapeshift/web/pulls) for the same update/change?

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [x] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

N/A

## Testing

- We still wait ~17 seconds after Tx confirmation before showing the confirmed screen
- Closing the status modal doesn't rug us from refetching savers user data after it is updated in the savers API
- DeFi action buttons for savers are disabled while there are any pending queries

### Engineering

- ^

### Operations

- ^

### CODE REVIEW

**Comment by gomesalexandre** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #3720** <a href="https://app.graphite.dev/github/pr/shapeshift/web/3720" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/shapeshift/web/3720?utm_source=stack-comment).

File: src/context/TransactionsProvider/TransactionsProvider.tsx [hunk 1]
```diff
@@ -49,7 +56,30 @@ export const TransactionsProvider: React.FC<TransactionsProviderProps> = ({ chil
...
+      if (status === TxStatus.Confirmed) return
+
+      if (
+        isSupportedThorchainSaversChainId(chainId) &&
+        transfers.some(({ assetId }) => isSupportedThorchainSaversAssetId(assetId))
+      ) {
+        const { getOpportunitiesUserData } = opportunitiesApi.endpoints
+        // Artificial longer completion time, since THORChain Txs take around 15s after confirmation to be picked in the API
+        // This way, we ensure "View Position" actually routes to the updated position
+        new Promise(resolve => setTimeout(resolve, 17000)).then(() => {
```
Comment (0xdef1cafe) [inline src/context/TransactionsProvider/TransactionsProvider.tsx:69]: how do you feel about a lil util fn so we can `await sleep(17000)` like we're python kids?

Comment (gomesalexandre) [reply src/context/TransactionsProvider/TransactionsProvider.tsx]: [here_we_go](9d88bed8e147bae9f75315e13f61b708f9e70cc0)

File: src/features/defi/providers/thorchain-savers/components/ThorchainSaversManager/Overview/ThorchainSaversOverview.tsx [hunk 2]
```diff
@@ -203,8 +228,15 @@ export const ThorchainSaversOverview: React.FC<OverviewProps> = ({
     return makeDefaultMenu({
       isFull: opportunityMetadata?.isFull,
       hasPendingTxs,
+      hasPendingQueries: Boolean(hasPendingQueries),
```
Comment (0xdef1cafe) [inline src/features/defi/providers/thorchain-savers/components/ThorchainSaversManager/Overview/ThorchainSaversOverview.tsx:231]:
```suggestion
      hasPendingQueries,
```

Comment (gomesalexandre) [reply src/features/defi/providers/thorchain-savers/components/ThorchainSaversManager/Overview/ThorchainSaversOverview.tsx]: https://github.com/shapeshift/web/pull/3720/commits/189833ee1a11fcc584c6d144c0462998be2f414a

**Review by 0xdef1cafe — APPROVED** [review]

**Pull request merged**